lib/authorization.rb

```diff
@@ -79,15 +79,27 @@ def self.matched_policies_dataset(project_id, subject_id, actions = nil, object_
         object_id = UBID.parse(object_id).to_uuid
       end
 
-      if (klass = UBID.class_for_ubid(ubid))
+      klass = UBID.class_for_ubid(ubid)
+      klass = ApiKey if ubid.start_with?("et")
```

Comment:
I recall seeing multiple special checks for `ApiKey`'s UBID while reviewing your authorization v2 PRs. Should we consider using a distinct UBID type for the ApiKey instead of etc, to eliminate these specific checks?

Reply:
The issue there is all current ApiKey values use the ETC ubid type.  We could make new ApiKey values use a different ubid type, but unless we want to invalidate the current ones, we need to keep this support.

Reply:
That approach would work.